### PR TITLE
feat(collections): pinboard utility parity with strict notes boundary

### PR DESCRIPTION
**Pull request by rmusser01**
## Summary
- implement Pinboard-inspired utility upgrades for collections without crossing strict notes-module boundaries
- add reading saved-search CRUD surfaces (backend + frontend + tests)
- add reading item-to-note link endpoints/UI with strict same-user note existence checks
- add archive-mode controls and reading metadata enhancements
- guard new surfaces behind feature flags and wire endpoint checks

## Validation
- `python -m pytest tldw_Server_API/tests/Collections/test_reading_api.py tldw_Server_API/tests/Collections/test_reading_service.py tldw_Server_API/tests/Collections/test_reading_saved_searches_db.py tldw_Server_API/tests/Collections/test_reading_note_links_db.py -v`
- `python -m pytest tldw_Server_API/tests/Collections/test_reading_api.py -k "saved_searches or note_link or archive_requested" -v`
- `bunx vitest run apps/packages/ui/src/components/Option/Collections/ReadingList/__tests__/saved-searches-menu.test.tsx apps/packages/ui/src/components/Option/Collections/ReadingList/__tests__/reading-item-detail-note-links.test.tsx apps/packages/ui/src/components/Option/Collections/ReadingList/__tests__/reading-list-page.archive-mode.test.tsx apps/packages/ui/src/services/tldw/__tests__/reading-saved-searches.test.ts`
- `python -m bandit -r tldw_Server_API/app/api/v1/endpoints/collections.py tldw_Server_API/app/core/Collections tldw_Server_API/tests/Collections -f json -o /tmp/bandit_collections_pinboard.json` (0 findings)

## Summary by CodeRabbit

* **Documentation**
  * Updated product requirements to define two new Reading features: reusable saved searches that persist list filters and sorting preferences.
  * Added specifications for reading item-to-note linking while maintaining Notes as the primary note management system.

### CODE REVIEW

File: tldw_Server_API/app/core/Collections/reading_service.py [hunk 1]
```diff
@@ -325,6 +351,74 @@ async def _ingest_non_html(
...
+        if READING_ARCHIVE_MAX_BYTES > 0 and len(content_bytes) > READING_ARCHIVE_MAX_BYTES:
+            return None, "archive_too_large"
+
+        timestamp = datetime.now(tz=timezone.utc).strftime("%Y%m%d_%H%M%S")
+        safe_title = _safe_filename_fragment(title, max_len=40)
+        filename = f"reading_archive_{item_id}_{safe_title}_{timestamp}.md"
+        outputs_dir = DatabasePaths.get_user_outputs_dir(self.user_id)
+        outputs_dir.mkdir(parents=True, exist_ok=True)
+        path = outputs_dir / filename
+        path.write_text(content, encoding="utf-8")
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_Server_API/app/core/Collections/reading_service.py]: Addressed in fa903d8da. Archive file-system operations now use await asyncio.to_thread(...) for both directory creation and file writes, so the async save path avoids direct blocking FS calls. Added regression coverage in test_archive_creation_uses_asyncio_to_thread_for_fs_io.

File: Docs/Product/Completed/Content_Collections_UX_Backlog_PRD.md [hunk 2]
```diff
@@ -59,6 +61,16 @@ This PRD tracks the remaining UX delivery work for Content Collections. It rolls
...
+- Reading saved searches use:
+  - `POST /api/v1/reading/saved-searches`
+  - `GET /api/v1/reading/saved-searches`
+  - `PATCH /api/v1/reading/saved-searches/{search_id}`
+  - `DELETE /api/v1/reading/saved-searches/{search_id}`
+- Reading item-note links use:
+  - `POST /api/v1/reading/items/{item_id}/links/note`
+  - `GET /api/v1/reading/items/{item_id}/links`
+  - `DELETE /api/v1/reading/items/{item_id}/links/note/{note_id}`
+  - Strict boundary rule: note content lifecycle remains in `/api/v1/notes`; Reading can only link/unlink note IDs after same-user existence checks.
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply Docs/Product/Completed/Content_Collections_UX_Backlog_PRD.md:73]: Addressed in fa903d8da. Sections 7 and 8 in Content_Collections_UX_Backlog_PRD.md now include explicit acceptance and test bullets for reading saved-search endpoints and item-note link endpoints, including payload validation, response-code/error-path coverage, auth checks, edge cases, and the strict notes-boundary rule.

File: Docs/Plans/2026-03-02-collections-pinboard-strict-boundary-implementation.md [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+git add apps/packages/ui/src/services/tldw/TldwApiClient.ts
+git add apps/packages/ui/src/services/tldw/__tests__/reading-saved-searches.test.ts
+git add apps/packages/ui/src/store/collections.tsx
+git add apps/packages/ui/src/components/Option/Collections/ReadingList/AddUrlModal.tsx
+git add apps/packages/ui/src/components/Option/Collections/ReadingList/ReadingItemsList.tsx
+git add apps/packages/ui/src/components/Option/Collections/ReadingList/SavedSearchesMenu.tsx
+git add apps/packages/ui/src/components/Option/Collections/ReadingList/__tests__/saved-searches-menu.test.tsx
+git add apps/packages/ui/src/components/Option/Collections/ReadingList/ReadingItemDetail.tsx
+git add apps/packages/ui/src/components/Option/Collections/ReadingList/__tests__/reading-item-detail-note-links.test.tsx
+git commit -m "feat(collections): deliver pinboard utility parity with strict notes boundary"
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply Docs/Plans/2026-03-02-collections-pinboard-strict-boundary-implementation.md:508]: Addressed in fa903d8da. The Step 5 commit block was rewritten into grouped, multiline git add commands to make the instructions shorter and easier to scan while keeping explicit file-level staging for this scoped change set.

File: tldw_Server_API/app/core/Collections/reading_service.py [hunk 4]
```diff
@@ -29,6 +34,8 @@
 from tldw_Server_API.app.core.Web_Scraping.url_utils import normalize_for_crawl
 
 READING_DEFAULT_STATUS = "saved"
+READING_ARCHIVE_MAX_BYTES = int(os.getenv("READING_ARCHIVE_MAX_BYTES", str(5 * 1024 * 1024)) or str(5 * 1024 * 1024))
+READING_ARCHIVE_RETENTION_DAYS = int(os.getenv("READING_ARCHIVE_RETENTION_DAYS", "30") or "30")
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_Server_API/app/core/Collections/reading_service.py]: Addressed in fa903d8da. Replaced direct int(os.getenv(...)) import-time parsing with _env_int(...), which validates values, logs invalid/out-of-range inputs, and falls back to safe defaults instead of crashing module import. Added regression coverage in test_archive_env_values_fallback_on_invalid.

File: tldw_Server_API/app/core/Collections/reading_service.py [hunk 5]
```diff
@@ -422,8 +524,35 @@ async def save_url(
...
+                archive_error = f"archive_failed:{exc}"
+            meta_patch: dict[str, object] = {"archive_requested": True}
+            if archive_output_id is not None:
+                meta_patch["has_archive_copy"] = True
+                meta_patch["archive_output_id"] = archive_output_id
+                meta_patch["archive_error"] = None
+            elif archive_error:
+                meta_patch["archive_error"] = archive_error
+            with contextlib.suppress(_READING_SERVICE_NONCRITICAL_EXCEPTIONS):
+                item_row = self.collections.update_content_item(item_row.id, metadata=meta_patch)
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_Server_API/app/core/Collections/reading_service.py]: Addressed in fa903d8da. Removed contextlib.suppress(...) around archive metadata updates and replaced it with explicit exception handling that logs the failure and propagates an archive_metadata_update_failed signal via archive_error. Added regression coverage in test_archive_metadata_update_failure_is_reported.
